gpflow/likelihoods.py

```diff
@@ -583,3 +583,136 @@ def conditional_variance(self, F):
         E_y = tf.matmul(phi, Ys)
         E_y2 = tf.matmul(phi, tf.square(Ys))
         return tf.reshape(E_y2 - tf.square(E_y), tf.shape(F))
+
+
+class MonteCarloLikelihood(Likelihood):
+    def __init__(self, name=None):
+        super().__init__(name)
+        self.num_monte_carlo_points = 100
+        del self.num_gauss_hermite_points
+
+    def _mc_evals(self, func, Fmu, Fvar, Y=None, epsilon=None):
+        N, D = tf.shape(Fmu)[0], tf.shape(Fvar)[1]
+        if epsilon is None:
+            epsilon = tf.random_normal((self.num_monte_carlo_points, N, D), dtype=settings.float_type)
+        mc_x = Fmu[None, :, :] + tf.sqrt(Fvar[None, :, :]) * epsilon
+        mc_Xr = tf.reshape(mc_x, (self.num_monte_carlo_points * N, D))
+
+        if Y is not None:
```

Comment:
Is this to support likelihoods which don't have Y?

Reply:
No, predict_mean_and_var() doesn't have Y